flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteInputChannel.java

```diff
@@ -170,7 +170,7 @@ void retriggerSubpartitionRequest(int subpartitionIndex) throws IOException {
 	@Override
 	Optional<BufferAndAvailability> getNextBuffer() throws IOException {
 		checkError();
-		checkState(partitionRequestClient != null, "Queried for a buffer before requesting a queue.");
+		checkState(partitionRequestClient != null, "Queried for a buffer before requesting a subpartition.");
```

Comment:
nit: If you are planning to keep this commit as an independent one and not be squashed with the bugfix, could you change it's commit message from:
> optimize error message

to:
> [FLINK-18348][hotfix] Improve error messages in RemoteInputChannel

? I'm also fine with squashing it - as you prefer.